### PR TITLE
Overseer integration

### LINKED ISSUE #241: Setup faucet funnel tracking
Most of the plumbing to send the events to overseer is in place already, what is missing is a reference id per user that persists throughout the signup steps.

The reference id should be assignable by querystring when initiating signup so that the funnel can include steps prior to signup

### PR DESCRIPTION
**Pull request by jnordberg**
Closes #241

### CODE REVIEW

**Comment by jnordberg** [comment]
What where the changes in api.js? Hard to tell with the reformat (my bad, should have fixed the prettier setting in a separate pr)

**Comment by jnordberg** [comment]
Can you give me a list of the step names in the order they would be called for a successful signup?

File: src/components/CreateAccount.js [hunk 1]
```diff
@@ -67,30 +67,33 @@ class CreateAccount extends Component {
...
+                                        ? 'username'
+                                        : 'password',
+                                stepNumber: data.username === '' ? 0 : 1,
+                                username: data.username,
+                                reservedUsername: data.reservedUsername,
+                                email: data.email,
+                                query: data.query,
+                                xref: data.xref,
+                            },
+                            logStep(data.xref, 'approved_user_username_step')
```
Comment (jnordberg) [inline src/components/CreateAccount.js]: This looks wrong, why is setState called with the output of logStep?

Comment (gl2748) [reply src/components/CreateAccount.js]: `setState` takes a callback as its second argument.

for comepleteness this callback should really be...
`logStep(this.state.xref, 'approved_user_username_step')`

Basically guarantee that the step is only logged after state has been set, and that the xref being logged is the same for both.

Comment (jnordberg) [reply src/components/CreateAccount.js]: You'll want to wrap that in a function block then, right now logStep is called before setState and the result (undefined) is passed as the callback

Comment (gl2748) [reply src/components/CreateAccount.js]: good point.  I just bumped it out of the setState callback it's not necessary.

**Comment by gl2748** [comment]
@jnordberg This branch has a clean commit history with the prettier run seperated from code changes.
https://github.com/steemit/faucet/compare/reverse-lint?expand=1

**Comment by gl2748** [comment]
@jnordberg I updated the comment above to describe how the steps are logged.

I am going to try and get this tested with overseer running locally today.

**Comment by jnordberg** [comment]
Ok, I'm still not 100% what the steps mean. `finish_step` for example, is that the last step?

**Comment by jnordberg** [comment]
Superseded by #313

**Pull request closed without merging**